dashboard/app/controllers/pd/application_dashboard_controller.rb

```diff
@@ -1,16 +1,26 @@
 module Pd
   class ApplicationDashboardController < ApplicationController
+    include Application::RegionalPartnerTeacherconMapping
+
+    TEACHERCON = "teachercon"
+    LOCAL_SUMMER = "local summer"
+
     authorize_resource class: 'Pd::Application::ApplicationBase'
 
     def index
       view_options(full_width: true)
 
       is_workshop_admin = current_user.permission? UserPermission::WORKSHOP_ADMIN
       regional_partners = is_workshop_admin ? RegionalPartner.all : current_user.regional_partners
+      regional_partners = regional_partners.select(:id, :name, :group).map do |partner|
+        processed_partner = partner.attributes
+        processed_partner["workshopType"] = get_matching_teachercon(partner) ? TEACHERCON : LOCAL_SUMMER
+        processed_partner
+      end
```

Comment:
This looks a bit cumbersome.

I'd prefer this logic, even though it's small, to be in the model or a serializer. This might be a good time to create a new serializer so we don't need to mess with the model attributes.

Ideally the workshop type would be stored in the DB for each partner, but for now we can use this `get_matching_teachercon` method internally, wrapped up a `workshop_type` method.